js/index.js

```diff
@@ -21,63 +21,148 @@ let state = {
+};
+const toggleVisibility = (style, state) =>
+  state ? (style.visibility = "visible") : (style.visibility = "hidden");
+
+const renderPepperoni = () =>
+  document.querySelectorAll(".pep").forEach(onePep => {
+    toggleVisibility(onePep.style, state.pepperoni);
   });
-}
 
-function renderMushrooms() {
+const renderMushrooms = () =>
   // Iteration 1: set the visibility of `<section class="mushroom">`
-}
+  document
+    .querySelectorAll(".mushroom")
+    .forEach(mushroom => toggleVisibility(mushroom.style, state.mushrooms));
 
-function renderGreenPeppers() {
+const renderGreenPeppers = () =>
   // Iteration 1: set the visibility of `<section class="green-pepper">`
-}
+  document
+    .querySelectorAll(".green-pepper")
+    .forEach(greenPepper =>
+      toggleVisibility(greenPepper.style, state.greenPeppers)
+    );
 
-function renderWhiteSauce() {
+const renderWhiteSauce = () =>
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
-}
+  document
+    .querySelectorAll(".sauce")
+    .forEach(sauce => sauce.classList.toggle("sauce-white", state.whiteSauce));
 
-function renderGlutenFreeCrust() {
+const renderGlutenFreeCrust = () =>
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
-}
+  document
+    .querySelectorAll(".crust")
+    .forEach(glutenFreeCrust =>
+      glutenFreeCrust.classList.toggle(
+        "crust-gluten-free",
+        state.glutenFreeCrust
+      )
+    );
+
+// Iteration 3: add/remove the class "active" of each `<button class="btn">`
+//Method 1: //Method 1 : using the state Object
+const toggleByClassName = (className, state) =>
+  document
+    .getElementsByClassName(className)[0]
+    .classList.toggle("active", state);
+
+const renderButtons1 = () => {
+  toggleByClassName("btn-pepperoni", state.pepperoni);
+  toggleByClassName("btn-mushrooms", state.mushrooms);
+  toggleByClassName("btn-green-peppers", state.greenPeppers);
+  toggleByClassName("btn-sauce", state.whiteSauce);
+  toggleByClassName("btn-crust", state.glutenFreeCrust);
+};
+//Method 2 : using the combined state/className Object
+const ClassNameByStatus = {
+  pepperoni: "pepperoni",
+  mushrooms: "mushrooms",
+  greenPeppers: "green-peppers",
+  whiteSauce: "sauce",
+  glutenFreeCrust: "crust"
+};
+const toggleByState = btnElement => {
+  Object.keys(state).forEach(stateKey => {
+    if (btnElement.className.includes(ClassNameByStatus[stateKey]))
+      btnElement.classList.toggle("active", state[stateKey]);
+  });
+};
+
+const renderButtons = () => {
+  document
+    .querySelectorAll(".panel.controls button")
+    .forEach(button => toggleByState(button));
+};
 
-function renderButtons() {
-  // Iteration 3: add/remove the class "active" of each `<button class="btn">`
-}
+const createIngredientElement = ingredient => {
+  const newIngredient = document.createElement("li");
+  newIngredient.innerHTML = `$${ingredient.price} ${ingredient.name}`;
+  return newIngredient;
+};
```

Comment:
nice approach!
As an example, there is also this approach that is shorter and cleaner.

`function renderButtons() { var ingArr = Object.keys(state); document.querySelectorAll(".btn").forEach(function(item, index) { if (state[ingArr[index]]) { item.classList.add("active"); } else { item.classList.remove("active"); } }); }`